chain/params.go

```diff
@@ -1,18 +1,54 @@
 package chain
 
 import (
+	"errors"
 	"math/big"
+	"sort"
+	"strconv"
 
 	"github.com/0xPolygon/polygon-edge/types"
 )
 
+var (
+	// ErrBurnContractAddressMissing is the error when a contract address is not provided
+	ErrBurnContractAddressMissing = errors.New("burn contract address missing")
+)
+
 // Params are all the set of params for the chain
 type Params struct {
 	Forks          *Forks                 `json:"forks"`
 	ChainID        int                    `json:"chainID"`
 	Engine         map[string]interface{} `json:"engine"`
 	Whitelists     *Whitelists            `json:"whitelists,omitempty"`
 	BlockGasTarget uint64                 `json:"blockGasTarget"`
+
+	// Governance contract where the token will be sent to and burn in london fork
+	BurnContract map[string]string `json:"burnContract"`
+}
+
+// CalculateBurnContract calculates burn contract address for the given block number
+func (p *Params) CalculateBurnContract(block uint64) (types.Address, error) {
+	keys := make([]string, 0, len(p.BurnContract))
+	for k := range p.BurnContract {
+		keys = append(keys, k)
+	}
+
+	if len(keys) == 0 {
+		return types.ZeroAddress, ErrBurnContractAddressMissing
+	}
+
+	sort.Strings(keys)
+
+	for i := 0; i < len(keys)-1; i++ {
+		valUint, _ := strconv.ParseUint(keys[i], 10, 64)
+		valUintNext, _ := strconv.ParseUint(keys[i+1], 10, 64)
+
+		if block > valUint && block < valUintNext {
```

Comment:
Just a question:
Shouldn't the condition be inclusive on one side?
If, for example,  keys are `[1, 10, 20, 30]` and block number is 10, than the burn contract for 30 will be returned
It seams to me that first part of condition should be `block >= valUint`

Reply:
Good point. There was a bug because there was a string sorting but not int which is wrong. Fixed.